modules/core/src/main/java/org/apache/ignite/internal/processors/service/GridServiceProxy.java

```diff
@@ -354,6 +358,42 @@ T proxy() {
         return proxy;
     }
 
+    /**
+     * Calls service method, measures and registers its duration.
+     *
+     * @param srvcProc Current service processor.
+     * @param srvc The service object.
+     * @param srvcName The service name.
+     * @param mtd Method to call.
+     * @param args Arguments for {@code mtd}.
+     */
+    private static Object measureServiceMethod(
+        ServiceProcessorAdapter srvcProc,
+        Service srvc,
+        String srvcName,
+        Method mtd,
+        Object[] args
+    ) throws InvocationTargetException, IllegalAccessException {
+
+        long startTime = System.nanoTime();
+
+        try {
+            return mtd.invoke(srvc, args);
+        }
+        finally {
+            if (srvcProc instanceof IgniteServiceProcessor) {
+                long duration = System.nanoTime() - startTime;
+
+                HistogramMetricImpl histogram = ((IgniteServiceProcessor)srvcProc).histogram(srvcName, mtd);
+
+                assert histogram != null;
```

Comment:
Do you realy want this assertion? When assertions are enabled, this can create some problems.